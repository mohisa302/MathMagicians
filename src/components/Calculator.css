input {
  margin: 0;
  padding: 0;
  outline: none;
  color: black;
  font-size: x-large;
  font-weight: bold;
  outline: none;
  outline: none;
  border-top-style: hidden;
  border-right-style: hidden;
  border-left-style: hidden;
  border-bottom-style:hidden;
  border: 1px gray solid;
}

.row {
  display: flex;
  row-gap: 0;
  column-gap: 0;
}

.input-text {
  background-color: rgb(145, 144, 144);
  color: white;
  width: 100%;
  height: 10vh;
  text-align: end;
}

.container {
  display: flex;
  flex-direction: column;
  padding: 0;
  margin: 0;
  width: 40vw;
  flex-wrap: wrap;
  align-self: center;
}

td {
  display: flex;
  width: 100%;
  padding: 0;
}

.btn {
  display: flex;
  justify-content: center;
  background-color: rgb(218, 221, 223);
  height: 6rem;
  flex-grow: 1;
}

.op {
  background-color: rgb(238, 154, 69);
}

div {
  display: flex;
  justify-content: center;
  /* justify-content: flex-end; */
  /* height: 80vh;
  width: 100vw; */
}

.zero {
  width: 200%;
}

table {
  margin: 0;
}