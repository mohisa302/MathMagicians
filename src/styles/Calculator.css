input {
  margin: 0;
  padding: 0;
  outline: none;
  color: black;
  font-size: x-large;
  font-weight: bold;
  border-top-style: hidden;
  border-right-style: hidden;
  border-left-style: hidden;
  border-bottom-style: hidden;
}

.row {
  display: flex;
  row-gap: 0;
  column-gap: 0;
}

.input-text {
  display: flex;
  color: white;
  width: 100%;
  height: 10vh;
  align-items: center;
  justify-content: flex-end;
  font-size: x-large;
  font-weight: bold;
  background-color: gray;
}

.container {
  display: flex;
  flex-direction: column;
  padding: 0;
  margin: 0;
  width: 40vw;
  flex-wrap: wrap;
  align-self: center;
}

td {
  display: flex;
  width: 100%;
  padding: 0;
}

.btn {
  display: flex;
  justify-content: center;
  background-color: rgb(218, 221, 223);
  height: 6rem;
  flex-grow: 1;
  border: 1px gray solid;
}

.op {
  background-color: rgb(238, 154, 69);
}

.zero {
  width: 200%;
}

table {
  margin: 0;
}

.mainContainer {
  display: flex;
  justify-content: space-between;
}
