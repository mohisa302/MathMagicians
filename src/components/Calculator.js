import React, { useState } from 'react';
import '../styles/Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: null, operation: null });

  const clickLisetener = (event) => {
    setState(calculate(state, event.target.value));
  };
  const { total, operation, next } = state;
  return (
    <table className="container">
      <tr className="row">
        <td>
          <div className="input-text">
            {total}
            {operation}
            {next}
          </div>
        </td>
      </tr>
      <tr className="row">
        <td>
          <input type="button" value="AC" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="+/-" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="%" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value={'\u00F7'} className="btn op" onClick={clickLisetener} />
        </td>
      </tr>
      <tr className="row">
        <td>
          <input type="button" value="7" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="8" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="9" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="x" className="btn op" onClick={clickLisetener} />
        </td>
      </tr>
      <tr className="row">
        <td>
          <input type="button" value="4" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="5" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="6" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="-" className="btn op" onClick={clickLisetener} />
        </td>
      </tr>
      <tr className="row">
        <td>
          <input type="button" value="1" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="2" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="3" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="+" className="btn op" onClick={clickLisetener} />
        </td>
      </tr>
      <tr className="row">
        <td className="zero">
          <input type="button" value="0" className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="." className="btn" onClick={clickLisetener} />
        </td>
        <td>
          <input type="button" value="=" className="btn op" onClick={clickLisetener} />
        </td>
      </tr>
    </table>
  );
};

export default Calculator;
