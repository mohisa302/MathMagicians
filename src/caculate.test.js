import * as React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from './components/Calculator';
import {fireEvent } from '@testing-library/react'
import calculate from './logic/calculate';
import operate from './logic/operate';

describe('Calculator', () => {
  it('renders Calculator component', () => {
    render(<Calculator />);
    const linkElement = screen.getByText(/do some math/i);
    expect(linkElement).toBeInTheDocument();
  });

  it('check one of the calculator button exists', () => {
    render(<Calculator />);
    const btn = screen.getByDisplayValue('1');
    expect(btn).toBeInTheDocument();
  });

  it('check calculator\'s input exists', () => {
    render(<Calculator />);
    const input = screen.getByTestId('input-cal-text');
    expect(input).toHaveTextContent('0');
  });
    
  it(`check aftre push number btn input change`, () => {
    render(<Calculator />);
    const eightBtn = screen.getByDisplayValue('8')
    fireEvent.click(eightBtn);
    expect(eightBtn).toHaveValue('8');
  });  

  test('Builds the snapchot of the Calculator component safely', () => {
    const rend = renderer.create(<Calculator />);
    expect(rend.toJSON()).toMatchSnapshot();
  });
  
  test('Render Calculator component', () => {
    render(<Calculator />);
  });

  const obj = { total: null, next: null, operation: null };

  it('Return null object if it is AC', () => {
    const result = calculate(obj, 'AC');
    expect(result).toEqual({ total: null, next: null, operation: null });
  });

  it('Test for modulus operation', () => {
    expect(operate(2, 5, '%')).toBe('2');
  });
});
