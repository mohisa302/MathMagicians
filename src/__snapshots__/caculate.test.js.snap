// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Builds the Calculator component safely 1`] = `
<div
  className="mainContainer"
>
  <h2>
    Let's do some math !
  </h2>
  <table>
    <tbody
      className="container"
    >
      <tr
        className="row"
      >
        <td>
          <div
            className="input-text"
            data-testid="input-cal-text"
          >
            0
          </div>
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="AC"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="+/-"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="%"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="÷"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="7"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="8"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="9"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="x"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="4"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="5"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="6"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="-"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="1"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="2"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="3"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td
          className="zero"
        >
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="0"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="."
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="="
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Calculator Builds the snapchot of the Calculator component safely 1`] = `
<div
  className="mainContainer"
>
  <h2>
    Let's do some math !
  </h2>
  <table>
    <tbody
      className="container"
    >
      <tr
        className="row"
      >
        <td>
          <div
            className="input-text"
            data-testid="input-cal-text"
          >
            0
          </div>
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="AC"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="+/-"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="%"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="÷"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="7"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="8"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="9"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="x"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="4"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="5"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="6"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="-"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="1"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="2"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="3"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td
          className="zero"
        >
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="0"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="."
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="="
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Calculator Builds the snapchot of the Header component safely 1`] = `
<header
  className="header"
>
  <h1>
    Math Magicians
  </h1>
  <nav>
    <ul
      className="navLinks"
    >
      <li>
        <a
          href="/MathMagicians/"
          onClick={[Function]}
        >
           Home | 
        </a>
      </li>
      <li>
        <a
          href="/MathMagicians/calculate"
          onClick={[Function]}
        >
          Calculator | 
        </a>
      </li>
      <li>
        <a
          href="/MathMagicians/quotes"
          onClick={[Function]}
        >
            Quotes 
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`Calculator Builds the snapchot of the Quote component safely 1`] = `
<h3
  className="quoteText"
>
  Mathematices is not about numbers equtions or algorithms: it is about undertanding William Paoul Thurston
</h3>
`;

exports[`Calculator Builds the snapchot of the home component safely 1`] = `
Array [
  <h2
    className="homeTitle"
  >
    welcome to our page  !
  </h2>,
  <div
    className="homeBody"
  >
    <p>
      simply dummy text of the printing and typesetting industry. has been the industrys standard dummy text ever since the 1500s when an unknown printer took a galley of type and scrambled it to make a type specimen book.
    </p>
    <br />
    <p>
      It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
    </p>
  </div>,
]
`;

exports[`Calculator Matches DOM Snapshot 1`] = `
<div
  className="mainContainer"
>
  <h2>
    Let's do some math !
  </h2>
  <table>
    <tbody
      className="container"
    >
      <tr
        className="row"
      >
        <td>
          <div
            className="input-text"
            data-testid="input-cal-text"
          >
            0
          </div>
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="AC"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="+/-"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="%"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="÷"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="7"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="8"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="9"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="x"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="4"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="5"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="6"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="-"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="1"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="2"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="3"
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="+"
          />
        </td>
      </tr>
      <tr
        className="row"
      >
        <td
          className="zero"
        >
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="0"
          />
        </td>
        <td>
          <input
            className="btn"
            onClick={[Function]}
            type="button"
            value="."
          />
        </td>
        <td>
          <input
            className="btn op"
            onClick={[Function]}
            type="button"
            value="="
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
